docs/syntax-jsx-support.md

```diff
@@ -0,0 +1,338 @@
+- **Minimize grammar ambiguity** with existing and future Luau syntax.
+
+### Non-goals (initially)
+
+These are intentionally excluded from the initial proposal to reduce ambiguity and implementation risk:
+
+- Raw text children (e.g. `<Foo>hello</Foo>`)
+- Fragments (e.g. `<>...</>`)
+- Spread attributes (e.g. `<Foo {...props} />`)
+- Namespaced attributes (e.g. `on:click=...`)
+- JSX comments syntax
+
+These can be added via follow-on RFCs once the core parsing approach proves robust.
+
+### Opt-in surface area
+
+**File extension**: `.luax` indicates JSX may appear in the file.
+
+**Feature flag**: The syntax must be gated by a flag (e.g. `FFlag::LuauJsx`) so it can ship experimentally and be rolled out carefully.
+
+### Syntax
+
+When enabled, an additional `simpleexp` form is introduced:
+
+- `jsxElement ::= '<' jsxTagName jsxAttributes ( '/>' | '>' jsxChildren '</' jsxTagName '>' )`
+- `jsxTagName ::= Name { '.' Name }`
+- `jsxAttributes ::= { jsxAttribute }`
+- `jsxAttribute ::= Name [ '=' ( String | '{' expr '}' ) ]`
+- `jsxChildren ::= { jsxChild }`
+- `jsxChild ::= jsxElement | '{' expr '}'`
+
+Notes:
+
+- `{ expr }` is used for embedding arbitrary Luau expressions in children and attribute values.
+- Attribute shorthand (`<Foo disabled />`) is allowed and is equivalent to `disabled={true}`.
+
+### Desugaring
+
+JSX constructs are desugared in the parser into an ordinary Luau function call:
+
+#### Tag lowering
+
+- If the tag begins with a lowercase identifier and has no dots (e.g. `<frame />`), it lowers to a string tag: `"frame"`.
+- Otherwise it lowers to an identifier/dotted expression (e.g. `<Foo.Bar />` lowers to `Foo.Bar`).
+
+#### Props lowering
+
+Attributes lower into a table literal (record fields). If there are no attributes, props lower to `nil`.
+
+#### Children lowering
+
+Children lower into additional positional arguments after `(tag, props)`.
+
+#### Examples
+
+Self-closing:
+
+```lua
+return <Foo bar={x} />
+```
+
+Desugars to (conceptually):
+
+```lua
+return jsx(Foo, { bar = x })
+```
+
+Lowercase tag:
+
+```lua
+return <frame />
+```
+
+Desugars to:
+
+```lua
+return jsx("frame", nil)
+```
+
+Nested child:
+
+```lua
+return <Foo><Bar /></Foo>
+```
+
+Desugars to:
+
+```lua
+return jsx(Foo, nil, jsx(Bar, nil))
+```
+
+Boolean attribute shorthand:
+
+```lua
+return <Foo disabled />
+```
+
+Desugars to:
+
+```lua
+return jsx(Foo, { disabled = true })
+```
+
+### Runtime / library integration
+
+Luau does **not** include React/Roact. JSX desugars to a call to a function named `jsx`. Projects can provide this in whichever way fits their runtime:
+
+- `jsx = React.createElement`
+- `jsx = Roact.createElement`
+- `jsx = MyCustomElementFactory`
+
+This keeps the language neutral and allows different runtimes to adopt the syntax without coupling Luau to any one library.
+
+#### Intrinsics mapping for lowercase tags
```

Comment:
This whole section feels like a direct argument for why the split behavior in your "tag lowering" section is a mistake. One piece of the design is creating a new problem that the other part solves. We should simply not create the problem.